src/sage/misc/temporary_file.py

```diff
@@ -533,14 +535,15 @@ def spyx_tmp() -> str:
     We cache the result of this function "by hand" so that the same
     temporary directory will always be returned. A function is used to
     delay creating a directory until (if) it is needed. The temporary
-    directory is removed when sage terminates by way of an atexit
-    hook.
+    directory is automatically removed when sage terminates.
     """
     global _spyx_tmp
     if _spyx_tmp:
         return _spyx_tmp
 
-    d = tempfile.TemporaryDirectory()
-    _spyx_tmp = os.path.join(d.name, 'spyx')
-    atexit.register(lambda: d.cleanup())
+    # We don't use `tempfile.TemporaryDirectory()` here because it
+    # will be cleaned up on child exit (e.g. for parallel testing)
+    # For some reason this doesn't affect the `TemporaryDirectory`
```

Comment:
Using `tempfile.TemporaryDirectory()` works just fine for me, so this comment may not be completely correct (however, it is not cleaned up on exit)

Reply:
Only on `spyx_tmp()`. The `TMP_DIR_FILENAME_BASE` one is cleaned up correctly

Reply:
So, there seems to be something fragile about using `TemporaryDirectory()` on a local variable that is meant to survive its function. There are no clear rules about when this object will be destroyed (apparently the `lambda: d.cleanup()` grabs a reference that stays in the atexit hook, but I'm not sure if the lambda takes a strong reference).

I guess the "right" way to acomplish what was intended here might be to store `d` in the global variable, and not just `d.name`.